lib/components/base-components/PrimitiveComponent/PrimitiveComponent.ts

```diff
@@ -87,6 +87,32 @@ export abstract class PrimitiveComponent<
     return undefined // Return undefined if not found
   }
 
+  getInheritedMergedProperty(propertyName: string): any {
+    const parentValue = this.parent?.getInheritedMergedProperty?.(propertyName)
+    const currentValue =
+      this._parsedProps?.[propertyName as keyof z.infer<ZodProps>]
+
```

Comment:
NEVER use the word value!!! It is against our code guidelines

Also don't use `current` inappropriately, `current` has to do with time or process.

You can spread null, i.e. `{ ...null, ...{a : 1 } }` = `{a: 1}`

```suggestion
  return { ...parentPropertyObject, ...myPropertyObject }
```